<?xml version="1.0"?>

<robot name="mim_diffdrive_arm" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <xacro:include filename="$(find mim_robots)/urdf/diffdrive/mim_diffdrive.urdf"/>

  <xacro:arg name="dummy_link" default="dummy_link"/>

  <link name="$(arg dummy_link)"></link>
  <joint name="dummy_to_car" type="fixed">
    <origin
      xyz="0 0 0"
      rpy="0 0 0" />
    <parent
      link="$(arg dummy_link)" />
    <child
      link="base_link" />
    <axis
      xyz="0 0 0" />
  </joint>

  <xacro:include filename="$(find mim_robots)/urdf/arm/mim_arm.urdf"/>

  <joint name="car_to_arm" type="fixed">
    <origin
      xyz="-0.2 0 0.321"
      rpy="0 0 0" />
    <parent
      link="base_link" />
    <child
      link="arm_base_link" />
    <axis
      xyz="0 0 0" />
  </joint>

  <xacro:include filename="$(find mim_robots)/urdf/gazebo/diffdrive_arm.gazebo"/>
  <xacro:include filename="$(find mim_robots)/urdf/diffdrive/diffdrive_transmission.urdf"/>
  <xacro:include filename="$(find mim_robots)/urdf/arm/arm_transmission.urdf"/>
</robot>
